compiler/src/org.graalvm.compiler.asm.amd64/src/org/graalvm/compiler/asm/amd64/AMD64Assembler.java

```diff
@@ -1353,6 +1354,7 @@ public void emit(AMD64Assembler asm, AVXSize size, Register dst, Register src1,
             asm.emitModRM(dst, src2);
```

Comment:
Regarding the caller of this method, we have to make sure the ordering of `src1` and `src2`. I think at `AMD64NodeMatchRules.logicalAndNot` we swap the operand ordering and result in the wrong output.